templates/users.html

```diff
@@ -0,0 +1,108 @@
+{% extends "layout.html" %}
+{% block pagetitle %}List of Users{% endblock %}
+{% block additionalpageheader %}
+    <script>
+        $(document).ready(function () {
+            var errorPopup = $('#errorPopup');
+
+            {% with messages = get_flashed_messages(with_categories=true) %}
+                {% if messages %}
+                    {% for category, message in messages %}
+                        {% if category == 'error' %}
+                            $('#errorMessage').text('{{ message }}');
+                            errorPopup.modal('show');
+                        {% endif %}
+                    {% endfor %}
+                {% endif %}
+            {% endwith %}
+
+            $('#okButton').click(function () {
+                errorPopup.modal('hide');
+            });
+        });
+    </script>
+    <style>
+        a {
+            text-decoration: none;
+            color: black;
+        }
+    </style>
+{% endblock %}
+{% block pagecontent %}
+    <div class="container">
+        <h2>List of Users</h2>
+        <hr>
+        <table class="table table-hover">
+            <thead>
+            <tr>
+                <th>Last Name</th>
+                <th>First Name</th>
+                <th>Email</th>
+                <th>Organization Code</th>
+                {% if user_type == 'researcher' %}
+                    <th>Status</th>
+                    <th>Max load</th>
+                {% endif %}
+                {% if user_type == 'archived' %}
+                    <th>Type</th>
+                {% endif %}
+                <th></th>
+            </tr>
+            </thead>
+            <tbody>
+            {% for user in users %}
+                <tr>
+                    <td><a href="{{ url_for("user.user_profile", user_id=user.id, current_year=dynamic_year) }}">
```

Comment:
Where is `dynamic_year` populated from ? I don't find its definition.

Reply:
dynamic_year is defined in the @app.context_processor